x-pack/plugin/mapper-match-only-text/src/main/java/org/elasticsearch/xpack/matchonlytext/mapper/MatchOnlyTextFieldMapper.java

```diff
@@ -0,0 +1,302 @@
+            throws IOException {
+            final Query query = textFieldType.phraseQuery(stream, slop, enablePosIncrements, queryShardContext);
+            return toQuery(query, queryShardContext);
+        }
+
+        @Override
+        public Query multiPhraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements, QueryShardContext queryShardContext)
+            throws IOException {
+            final Query query = textFieldType.multiPhraseQuery(stream, slop, enablePositionIncrements, queryShardContext);
+            return toQuery(query, queryShardContext);
+        }
+
+        @Override
+        public Query phrasePrefixQuery(TokenStream stream, int slop, int maxExpansions, QueryShardContext queryShardContext)
+            throws IOException {
+            final Query query = textFieldType.phrasePrefixQuery(stream, slop, maxExpansions, queryShardContext);
+            return toQuery(query, queryShardContext);
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
+            throw new IllegalArgumentException(CONTENT_TYPE + " fields do not support sorting and aggregations");
+        }
+
+    }
+
+    private final Builder builder;
+    private final FieldType fieldType;
+
+    private MatchOnlyTextFieldMapper(
+        String simpleName,
+        FieldType fieldType,
+        MatchOnlyTextFieldType mappedFieldType,
+        NamedAnalyzer indexAnalyzer,
+        MultiFields multiFields,
+        CopyTo copyTo,
+        Builder builder
+    ) {
+        super(simpleName, mappedFieldType, indexAnalyzer, multiFields, copyTo);
+        assert mappedFieldType.getTextSearchInfo().isTokenized();
+        assert mappedFieldType.hasDocValues() == false;
+        this.fieldType = fieldType;
+        this.builder = builder;
+    }
+
+    @Override
+    public FieldMapper.Builder getMergeBuilder() {
+        return new Builder(simpleName(), builder.indexCreatedVersion, builder.analyzers.indexAnalyzers).init(this);
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context) throws IOException {
+        final String value;
+        if (context.externalValueSet()) {
+            value = context.externalValue().toString();
+        } else {
+            value = context.parser().textOrNull();
+        }
+
+        if (value == null) {
+            return;
+        }
+
+        Field field = new Field(fieldType().name(), value, fieldType);
+        context.doc().add(field);
+        createFieldNamesField(context);
+    }
+
+    @Override
+    protected String contentType() {
+        return CONTENT_TYPE;
+    }
+
+    @Override
+    public MatchOnlyTextFieldType fieldType() {
+        return (MatchOnlyTextFieldType) super.fieldType();
+    }
+
+    @Override
+    protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
```

Comment:
I don't think you need to copy this bit, because there will be no BWC issue with a new mapper